pdm/formats/requirements.py

```diff
@@ -1,17 +1,28 @@
 import hashlib
 import urllib.parse
+from argparse import Namespace
+from os import PathLike
+from typing import Any, Dict, List, Optional, Tuple, Union
 
 from distlib.wheel import Wheel
+from pip._internal.index.package_finder import PackageFinder
+from pip._internal.req.req_install import InstallRequirement
```

Comment:
Import from `pdm.models.pip_shims` please

Reply:
fixed here

[`312287c` (#354)](https://github.com/pdm-project/pdm/pull/354/commits/312287c375db739caa180500afbc976002515113#diff-508798c1e0391fc888574533cc635c84f3be7f0a6732a439eafdc5bd2581da41R15)

and I checked all `pip._internal` importing not from `pdm.models.pip_shims`.

Some codes are still using `pip._vendor` such as `from pip._vendor.requests.models import Response`, and if you'd like to move all of them into pdm.models, pleate tell me.